### PR TITLE
feat: add read multiple instructions snippet

### PR DESCRIPTION
**Pull request by migui3230**

### CODE REVIEW

**Comment by jacobcreech** [comment]
@migui3230 Could you make the full source the actual program code? Want to be able to deploy this example if needed

File: docs/references/programs.md [hunk 1]
```diff
@@ -334,3 +334,31 @@ A basic instruction which initializes a hero state account, but with the above m
...
 </SolanaCodeGroup>
+
+## How to read multiple instructions from a transaction
+
+Since Solana allows us to take a peek at all of the instructions in the current transaction. We can store them in a variable and 
+iterate over them. We can do many things with this, like checking for suspicious transactions. 
+
+1. Read the number of instructions in the current transaction.
+2. Loop through the number of instructions
+3. Check for errors within the loop
```
Comment (jacobcreech) [inline docs/references/programs.md:345]: Let's remove this

File: docs/references/programs.md [hunk 2]
```diff
@@ -334,3 +334,31 @@ A basic instruction which initializes a hero state account, but with the above m
 
   </SolanaCodeGroupItem>
 </SolanaCodeGroup>
+
+## How to read multiple instructions from a transaction
+
+Since Solana allows us to take a peek at all of the instructions in the current transaction. We can store them in a variable and
```
Comment (jacobcreech) [inline docs/references/programs.md:340]:
```suggestion
Solana allows us to take a peek at all of the instructions in the current transaction. We can store them in a variable and 
```

**Review by jacobcreech — CHANGES_REQUESTED** [review]
A few changes + Full code requested

**Comment by jacobcreech** [comment]
@migui3230 Where is the actual code that inspects the instructions on the transaction in the program? I'm expecting to see something like 

```rust
let num_instructions = read_u16(&mut idx, &instruction_sysvar)
            .map_err(|_| ProgramError::InvalidAccountData)?;
```

and then iterating through those in the program. That would be the entire preview

**Comment by migui3230** [comment]
Gotcha, I understand now. Sorry for the confusion!

**Comment by migui3230** [comment]
@jacobcreech pushed changes im pretty sure this is what you're looking for?

**Pull request merged**